### PR TITLE
Electric Lighter

### PR DESCRIPTION
**Pull request by yay855**
## Summary

SUMMARY: [Content] "Added in new item, electric lighter, as well as a recipe for it and updated itemgroups to include it"

## Purpose of change

We have a crude electric lighter that you can craft and refillable gasoline lighters you can find, but no professionally-made electric lighters, which are a real thing IRL. Since firestarters aren't exactly hard to come by, this doesn't change the balance any, it's just a small QOL tidbit.

## Describe the solution

I added in a single item, an appropriate crafting recipe, and added it to itemgroups wherever refillable lighters can be found.

## Describe alternatives you've considered

It's such a small change, the only alternative I can think of is to not include it.

## Testing

Json seems to check out.

### CODE REVIEW

**Review by RoyalFox2140 — CHANGES_REQUESTED** [review]
image
image
@scarf005 We have 3 problems to address with this PR.
1. Electric firestarters and Electric lighters do not use charges. I don't know why, but it appears firestarters may have never used charge? Without charges this is infinite fire and that code fix is likely out of scope for this PR.

2. The spawn groups for lighters makes electric lighters rare and thus not spawn in many places, I only found them in pharmacies and storage units in testing. I think adding one or two at hardware stores, electronics stores, maybe camping or sports stores, would be a good addition but I don't see this as a blocker to the PR.

3. Neither the electric firestarter nor electric lighter has a defined movecost, I assume it defaults to a number as in testing time did pass, but I don't know what to do about that.

EDIT: My review was made on a version before Chorus added a movecost to both electric lighter.

**Review by scarf005 — COMMENTED** [review]
> Electric firestarters and Electric lighters do not use charges. I don't know why, but it appears firestarters may have never used charge? Without charges this is infinite fire and that code fix is likely out of scope for this PR.

this seems to happen when `max_charges` is not defined, weird.

**Comment by yay855** [comment]
What does max_charges do?

File: data/json/items/tool/fire.json [hunk 1]
```diff
@@ -261,5 +261,39 @@
...
     "flags": [ "FIRESTARTER" ]
+  },
+  {
+    "id": "electric_lighter",
+    "type": "TOOL",
+    "category": "tools",
+    "name": { "str": "electric lighter" },
+    "description": "A battery-powered lighter.  More efficient than anything you can make yourself, its batteries last for quite a while.",
+    "weight": "14 g",
+    "volume": "250 ml",
```
Comment (scarf005) [inline data/json/items/tool/fire.json:272]: it's a bit unwieldy considering regular lighter is 14ml, could i copy-from stats from lighter to make it lighter?

Comment (yay855) [reply data/json/items/tool/fire.json:272]: Yeah, probably a good idea.

Comment (RoyalFox2140) [reply data/json/items/tool/fire.json:272]: It has a light battery wouldn't it be limited by the size of the battery?

**Comment by scarf005** [comment]
> What does max_charges do?

https://docs.cataclysmbn.org/en/mod/json/reference/json_info/#tools

image

it says maximum charges a tool can hold, but the limit seems to be overridden for magazine inserted.

**Comment by yay855** [comment]
So then I'll just add in a base max_charges to it, make it 100, and then that's that.

File: data/json/items/tool/fire.json (hunk 1 above)
Comment (scarf005) [reply data/json/items/tool/fire.json:272]: https://github.com/scarf005/Cataclysm-BN/blob/651400f417fc3e54164d00411fd02da59404fd69/data/json/items/battery.json#L77-L85

even so, light battery is 25ml so 250ml would be very big. also, we could set [`magazine_well`](https://docs.cataclysmbn.org/ko/mod/json/reference/json_info/#generic-items) to make large batteries stick out.

**Comment by scarf005** [comment]
alright, i'll do the changes (and fix firestarter), please let me know if anything gets strange

File: data/json/items/tool/fire.json (hunk 1 above)
Comment (yay855) [reply data/json/items/tool/fire.json:272]: Oh, okay. Then yeah, let's do that.

**Comment by yay855** [comment]
> alright, i'll do the changes (and fix firestarter), please let me know if anything gets strange

Will do!

**Review by scarf005 — DISMISSED** [review]
https://github.com/cataclysmbnteam/Cataclysm-BN/assets/54838975/dbfdc5bc-a82e-4f57-9348-0817863d521c

so now i see, `max_charges` works as an internal charges even when it's magazine fed. it might feel 'too OP' to give both electric lighter and firestarter 100 internal charges. should i reduce it to 10 or 5 so it could be only used with magazine inserted?

**Review by scarf005 — APPROVED** [review]

**Pull request merged**